lms/djangoapps/teams/views.py

```diff
@@ -79,16 +81,18 @@ def has_team_api_access(user, course_key):
     Args:
       user (User): The user to check access for.
       course_key (CourseKey): The key to the course which we are checking access to.
+      access_username (string): If provided, access_username must match user.username for non staff access.
 
     Returns:
       bool: True if the user has access, False otherwise.
     """
-    return (CourseEnrollment.is_enrolled(user, course_key) or
-            CourseStaffRole(course_key).has_user(user) or
-            user.is_staff)
+    student_access = CourseEnrollment.is_enrolled(user, course_key)\
+        and (access_username == user.username if access_username else True)
+    course_staff_access = CourseStaffRole(course_key).has_user(user)
+    return student_access or course_staff_access or user.is_staff
```

Comment:
I wonder if we should write the code so that we don't need to do unnecessary checks (yes, I know this is premature optimization).

For example:

if user.is_staff return True

if CourseStaffRole(course_key).has_user(user) return True

then the student access logic (checking username first so that the course enrollment check is not necessary if self.username does not match.